### DISCUSSION TITLE
Get an action to open a registered page

### DISCUSSION THREAD

**Discussion by axelbdt** [filamentphp/filament#8205] [Help] [answered]
Hello,

Thanks for the great work on v3, it's really impressive and enjoyable so far.

I am afraid that this might be a XY question so here is a bit of context so you can let me know if it is.

I want two types of view on a Booking resource:
-  a detailed view that will show all fields like the default view
- a "coupon" view that is basically a summary of some field in the form a legally standardized text

I assumed the coupon view would be a ViewBookingCoupon page which I created as a ViewRecord page, but I am failing to have an action open this page.

I have registered it in addition to the default pages:

```php
    public static function getPages(): array
    {
        return [
            'index' => Pages\ListBookings::route('/'),
            'create' => Pages\CreateBooking::route('/create'),
            'view' => Pages\ViewBooking::route('/{record}'),
            'viewCoupon' => Pages\ViewBookingCoupon::route('/{record}/coupon'), // It's here
            'edit' => Pages\EditBooking::route('/{record}/edit'),
        ];
    }
```

How to idiomatically add an action that will  open this page ? What I tried ended up super convoluted code-wise so I scrapped it and am left with the following.

```php
            ->actions(
                [
                ActionGroup::make(
                    [
                    Tables\Actions\ViewAction::make(),
                    Tables\Actions\ViewAction::make('viewCoupon')
                        ->label(__('coupon')),
                    Tables\Actions\EditAction::make(),
                    Tables\Actions\DeleteAction::make(),
                            ]
                )
                    ], position: ActionsPosition::BeforeColumns
            )
```

Thanks in advance.

**Comment by danharrin** [comment, marked as answer]
`->url(fn (Booking $record): string => BookingResource::getUrl('viewCoupon', ['record' => $record]))`
